modules/fishing.py

```diff
@@ -80,44 +82,25 @@ async def _catch_fish(ctx):
 
 @bot.command()
 async def fish(ctx: Context):
-
-    new_fisher = True
-
-    for i in range(len(sessions)):
-        if ctx.author.id == sessions[i].id:
-            if not sessions[i].state == NOT_FISHING:
-                await ctx.send(f"{ctx.author.mention} You are already fishing.")
-                return
-
-            fisher = i
-            sessions[i].state = WAITING_FOR_BITE
-            new_fisher = False
-            break
-
-    if new_fisher:
-        fisher = len(sessions)
-        sessions.append(FishingSession)
-        sessions[fisher].id = ctx.author.id
-
-    sessions[fisher].state = WAITING_FOR_BITE
-    await ctx.send(f"{ctx.author.mention} You casts your line...")
-    await _bite_timer(ctx, fisher)
+    fisher = ctx.author.id
+    if fisher not in sessions:
+        sessions[fisher] = FishingState.WAITING_FOR_BITE
+        await ctx.send(f"{ctx.author.mention} You casts your line...")
+        await _bite_timer(ctx, fisher)
+    else:
+        await ctx.send(f"{ctx.author.mention} You are already fishing.")
 
 
 @bot.command()
 async def hook(ctx: Context):
+    fisher = ctx.author.id
 
-    for i in range(len(sessions)):
-        if ctx.author.id == sessions[i].id:
-            if sessions[i].state == WAITING_FOR_BITE:
-                await ctx.send(f"{ctx.author.mention} There was nothing on the line. Send **!fish** to recast.")
-                sessions[i].state = NOT_FISHING
-                return
-            elif sessions[i].state == NOT_FISHING or sessions[i].state == HOOKING:
-                await ctx.send(f"{ctx.author.mention} You are not fishing right now. Send **!fish** to start fishing.")
-                return
-            else:
-                sessions[i].state = NOT_FISHING
-                await _catch_fish(ctx)
-                return
-
+    if fisher not in sessions:
+        await ctx.send(f"{ctx.author.mention} You are not fishing right now. Send **!fish** to start fishing.")
+    else:
+        state = sessions[fisher]
+        if state == FishingState.WAITING_FOR_BITE:
+            await ctx.send(f"{ctx.author.mention} There was nothing on the line. Send **!fish** to recast.")
+        elif state == FishingState.FISH_BITING:
+            await _catch_fish(ctx)
+            del sessions[fisher]
```

Comment:
Same deal here as in `fish()`, just with slightly more complex state checking logic.